requests/utils.py

```diff
@@ -726,3 +726,11 @@ def urldefragauth(url):
     netloc = netloc.rsplit('@', 1)[-1]
 
     return urlunparse((scheme, netloc, path, params, query, ''))
+
+
+def determine_if_stream(data):
+    """Given data, determines if it should be sent as a stream.
+    """
+    is_iterable = hasattr(data, '__iter__')
```

Comment:
i am not sure if it is relevant in this case, but maybe using getattr is a better option here since hasattr has some issues in Python 2: https://hynek.me/articles/hasattr/